src/java.base/share/classes/java/security/PEMDecoder.java

```diff
@@ -43,96 +45,105 @@
+ * <ul>
+ *   <li>ENCRYPTED PRIVATE KEY : {@link PKCS8EncodedKeySpec}</li>
+ *   <li>ENCRYPTED PRIVATE KEY : {@link PublicKey} (if the encoding contains a public key)</li>
+ *   <li>ENCRYPTED PRIVATE KEY : {@link X509EncodedKeySpec} (if the encoding contains a public key)</li>
+ * </ul>
  *
  * <p> A new {@code PEMDecoder} instance is created when configured
- * with {@linkplain #withFactory(Provider)} and/or
- * {@linkplain #withDecryption(char[])}. {@linkplain #withFactory(Provider)}
- * configures the decoder to use only {@linkplain KeyFactory} and
- * {@linkplain CertificateFactory} instances from the given {@code Provider}.
- * {@linkplain #withDecryption(char[])} configures the decoder to decrypt all
- * encrypted private key PEM data using the given password.
- * Configuring an instance for decryption does not prevent decoding with
- * unencrypted PEM. Any encrypted PEM that fails decryption
- * will throw a {@link RuntimeException}. When an encrypted private key PEM is
- * used with a decoder not configured for decryption, an
- * {@link EncryptedPrivateKeyInfo} object is returned.
+ * with {@link #withFactory(Provider)} or {@link #withDecryption(char[])}.
+ * The {@link #withFactory(Provider)} method uses the specified provider
+ * to produce cryptographic objects from {@link KeyFactory} and
+ * {@link CertificateFactory}. The {@link #withDecryption(char[])} method configures the
+ * decoder to decrypt and decode encrypted private key PEM data using the given
+ * password.  If decryption fails, an {@link IllegalArgumentException} is thrown.
+ * If an encrypted private key PEM is processed by a decoder not configured
+ * for decryption, an {@link EncryptedPrivateKeyInfo} object is returned.
+ * A PEMDecoder configured for decryption will decode unencrypted PEM.
  *
- * <p>This class is immutable and thread-safe.
+ * <p> This class is immutable and thread-safe.
  *
- * <p> Here is an example of decoding a {@code PrivateKey} object:
+ * <p> Example: decode a private key:
  * {@snippet lang = java:
  *     PEMDecoder pd = PEMDecoder.of();
  *     PrivateKey priKey = pd.decode(priKeyPEM, PrivateKey.class);
  * }
  *
- * <p> Here is an example of a {@code PEMDecoder} configured with decryption
- * and a factory provider:
+ * <p> Example: configure decryption and a factory provider:
  * {@snippet lang = java:
  *     PEMDecoder pd = PEMDecoder.of().withDecryption(password).
- *         withFactory(provider);
- *     byte[] pemData = pd.decode(privKey);
+ *             withFactory(provider);
+ *     DEREncodable pemData = pd.decode(privKeyPEM);
  * }
  *
- * @implNote An implementation may support other PEM types and
- * {@code DEREncodable} objects. This implementation additionally supports
- * the following PEM types:  {@code X509 CERTIFICATE},
- * {@code X.509 CERTIFICATE}, {@code CRL}, and {@code RSA PRIVATE KEY}.
+ * @implNote This implementation decodes {@code RSA PRIVATE KEY} as {@code PRIVATE KEY},
```

Comment:
Shall we say "decodes as" or "treat the same as"? You decode "RSA PRIVATE KEY" to `PrivateKey` and not "PRIVATE KEY".